Odata-docs/client/v6/deal-with-server-driven-paging.md

```diff
@@ -19,15 +19,127 @@ Example:
 var context = new DefaultContainer(new Uri("https://services.odata.org/v4/TripPinServiceRW/"));
 
 // DataServiceQueryContinuation<T> contains the next link
-DataServiceQueryContinuation<Person> token = null;
+DataServiceQueryContinuation<Person> nextLink = null;
 
 // Get the first page
 var response = await context.People.ExecuteAsync() as QueryOperationResponse<Person>;
 
 // Loop if there is a next link
-while ((token = response.GetContinuation()) != null)
+while ((nextLink = response.GetContinuation()) != null)
 {
     // Get the next page
-    response = await context.ExecuteAsync<Person>(token) as QueryOperationResponse<Person>;
+    response = context.Execute<Person>(nextLink);
 }
 ```
+
+## Nested Pagination
+There are instances where we need to load pages of entities as well as pages of their related entities.
+The example below returns related Trips for each Person entity from the data service. It uses a do...while loop to paginate through Person entities and a nested while loop to paginate through the related Trips.
+We need to enumerate the response before calling `GetContinuation` on it.
+
+``` csharp
+var context = new DefaultContainer(new Uri("https://services.odata.org/v4/TripPinServiceRW/"));
+var pageCount = 0;
+var innerPageCount = 0;
+DataServiceQueryContinuation<Person> nextLink = null;
+
+try
+{
+    // Execute the query for all people and related trips,
+    // and get the response object.
+    var response = await
+        context.People.Expand("Trips")
+        .ExecuteAsync() as QueryOperationResponse<Person>;
+
+    // With a paged response from the service, use a do...while loop
+    // to enumerate the results before getting the next link.
+    do
+    {
+        // Write the page number.
+        Console.WriteLine("Person Page {0}:", ++pageCount);
+
+        // If nextLink is not null, then there is a new page to load.
+        if (nextLink != null)
+        {
+            // Load the new page from the next link URI.
+            response = await context.ExecuteAsync<Person>(nextLink)
```

Comment:
@KenitoInc I'm comparing this line with this [other one](https://github.com/MicrosoftDocs/OData-docs/pull/114/files#diff-f9bf5ec4bf534abf08213e96615c00dbR31) - Line 31 - and wondering why the latter uses `context.Execute<Person>(nextLink)` and not `await context.ExecuteAsync<Person>(nextLink)` like it is here

Reply:
#Resolved